### DISCUSSION TITLE
Installation files inside MSI

### DISCUSSION THREAD

**Discussion by bengtgunne** [wixtoolset/issues#7905] [Questions] [answered]
When I build my installation I want all files to end up inside the .msi file. How can I do this with WIX 4? In WIX 3 I used compressed="yes" for my components. For example:
### 
      <Component Id="cli_dbanalyzer.exe" Guid="{D38B7564-4C6C-49E8-9414-02F42BB17B0C}">
        <File Id="dbanalyzer.exe" KeyPath="yes" Vital="no" Compressed="yes" Source="$(var.distCur)\client\dbanalyzer.exe" />
      </Component>
With WIX 4 I get the error WIX0204: ICE35: File dbanalyzer.exe is marked compressed, but does not have a CAB specified in the Media table entry for its sequence number.

**Comment by chrpai** [comment, marked as answer]
Your Package element has to have a MediaTemplate child element with EmbedCab="yes" attribute. Here's an example from the IsWiX-Tutorials.

https://github.com/iswix-llc/iswix-tutorials/blob/master/WiX-v4-HeatWave/web-api/Installer/WebAPI/Product.wxs

This is also my number 1 liked answer on StackOverflow. It's funny how the simple things do this.

https://stackoverflow.com/questions/11284625/wix-single-msi-instead-of-msi-cab/11285208#11285208

**Comment by bengtgunne** [comment]
Thank you for answer! In my project file I was  using:
`<MediaTemplate EmbedCab="yes" />`
This did not work. However, the Stack overflow article suggested the following instead:
`<Media Id="1" Cabinet="CabFileName.cab" EmbedCab="yes" />`
And that worked for me!

**Reply by chrpai** [reply]
You don't need to put the Compressed attribute on every file element.  Remove them all and it will work without needing to specify the Id and Cabinet name on the Media element.

**Reply by bengtgunne** [reply]
When I remove the compressed attribute from the components, the installation files end up outside of the .msi file regardless if I have:
`<MediaTemplate EmbedCab="yes" />`
or
`<Media Id="1" Cabinet="CabFileName.cab" EmbedCab="yes" />`
or none of these.
